### DISCUSSION TITLE
[Resolvido] Execution of debuggers Finish in Few Seconds

### DISCUSSION THREAD

**Discussion by LeandroCGMS** [microsoft/debugpy#967] [General]
I've posted on 3 forums, over a month ago, but no one has shown to know the reason for the title problem. When I run the debugger through the VS Code client, it works normally, but only for a few seconds. I've already tweaked all my code due to changes from version 3 to version 4 of Django; I already changed the timeout to much longer; I already changed debugpy to ptvsd and went back to debugpy among other attempts.
![DockerDebugPyVSCode](https://user-images.githubusercontent.com/36311677/176787283-a997f610-f285-4729-a8e4-ad0050d72420.gif)
https://forum.djangoproject.com/t/debugpy-with-docker-and-visual-code-ending-automatically/14345

**Comment by int19h** [comment]
Some random thoughts from looking at the code.

You're using gunicorn, which generally implies gevent. For that to work with debugpy, it needs to be made aware of it from the moment it is loaded - and it must get loaded before gevent. In a "launch" scenario, you can just say `"gevent": true` to take care of this. But for "attach", you need to set an environment variable instead: `GEVENT_SUPPORT=True`.

**Reply by LeandroCGMS** [reply]
Below request, I put the 2 options, despite only the "env" option: {
           "GEVENT_SUPPORT": "True"
         }, be recommended for the "attach" scenario.
Even so, the problem remains.
I left it like this:
```
{
     "version": "0.2.0",
     "configurations": [
       {
         "name": "Run Django",
         "type": "python",
         "request": "attach",
         "env": {
           "GEVENT_SUPPORT": "True"
         },
         "gevent": true,
         "pathMappings": [
           {
             "localRoot": "${workspaceFolder}",
             "remoteRoot": "/usr/src/app"
           }
         ],
         "port": 8005,
         "host": "127.0.0.1",
       },
    ]
}
```

**Comment by int19h** [comment]
You cannot use "env" for "attach" - it only applies to "launch", where the debugger actually has the opportunity to change the environment before the app starts running. With "attach", you start the app yourself via gunicorn, so you'll need to change the environment through the same facility - presumably, in `docker-compose.yaml`.

**Reply by LeandroCGMS** [reply]
![image](https://user-images.githubusercontent.com/36311677/176686847-820fae25-a4ce-48e3-9f8b-031182bc3ae3.png)

I use the file version, 3.9, of docker-compose.yaml. Tried "GEVENT_SUPPORT: True", "GEVENT_SUPPORT: true", "GEVENT_SUPPORT: 'true'" and the problem remains. **Where am I wrong?**
____________________________________________________________________________________
PS: this way, as it is in the photo above, if I give an "echo "$GEVENT_SUPPORT"", the result is true, as in the photo below:
![image](https://user-images.githubusercontent.com/36311677/176786982-19adaa24-3305-4778-9407-79a1f52ff66d.png)

**Reply by LeandroCGMS** [reply]
Today, solving other problems, I came across a message, more or less like this, "The variable GEVENT_SUPPORT" is enabled, but GEVENT is not installed", so I added the dependency, remade the docker image and uploaded the container, with the docker-compose.yaml as shown in the image below, but even so, the problem remains.
![image](https://user-images.githubusercontent.com/36311677/177003988-36cd9622-6f4e-409d-93e8-565df9af74a0.png)

**Reply by LeandroCGMS** [reply]
Percebi que o problema também acontecia com o debugger "pdb", então deduzi que o mais provável, ser algo com o Gunicorn, logo, resolvi o problema acrescentando um parâmetro para ele usar um arquivo de configuração como mostrado na imagem abaixo:
![image](https://user-images.githubusercontent.com/36311677/178125086-b01c3647-4bdd-4aec-a2fe-16706edf73ac.png)
